Source/WebCore/xml/XPathFunctions.cpp

```diff
@@ -564,13 +565,9 @@ Value FunStringLength::evaluate() const
 
 Value FunNormalizeSpace::evaluate() const
 {
-    if (!argumentCount()) {
-        String s = Value(Expression::evaluationContext().node.get()).toString();
-        return s.simplifyWhiteSpace();
-    }
-
-    String s = argument(0).evaluate().toString();
-    return s.simplifyWhiteSpace();
+    // https://www.w3.org/TR/1999/REC-xpath-19991116/#function-normalize-space
+    String s = (!argumentCount() ? Value(Expression::evaluationContext().node.get()) : argument(0).evaluate()).toString();
```

Comment:
This seems way less readable than the original.

Reply:
I agree, I'd rater we keep the explicit `if` branch.

Reply:
Done.